algebra/src/linear_combination.rs

```diff
@@ -99,4 +99,109 @@ where
+mod test {
+    use rand::thread_rng;
+
+    use super::*;
+    use crate::{
+        fields::tweedle::Fr,
+        curves::tweedle::dee::DeeJacobian as TweedleDee, UniformRand, Group, GroupVec
+    };
+
+    const MAX_LENGTH: usize = 10;
+
+    /// Initialize a random LC of scalars and points and checks that the result of combine is indeed
+    /// given by (scalar_0 * point_0) + ... + (scalar_n-1 * point_n-1) 
+    fn test_scalar_point_random_lc_combine<G: Group>(scalars: Vec<G::ScalarField>, points: Vec<G>) {
+        let raw_lc = scalars
+            .clone()
+            .into_iter()
+            .zip(
+                points
+                    .iter()
+            )
+            .collect::<Vec<_>>();
+
+        // Initialize LC and combine
+        let lc = LinearCombination::new(raw_lc);
+        let combined_value = lc.combine();
+
+        // Compute the LC manually, instead, and check that the combined value is indeed the sum of the scalars times the points
+        let expected_combined_value = scalars
+            .into_iter()
+            .zip(
+                points
+                    .into_iter()
+            )
+            .map(|(scalar, point)| point * &scalar)
+            .fold(G::zero(), |acc, val| acc + val);
+
+        assert_eq!(combined_value, expected_combined_value);
+    }
+
+    /// Initialize a random LC from a single batching scalar and a vector of  points and checks that the result of combine is indeed
+    /// given by (0 * point_0) + (scalar * point_1) + (scalar^2 * point_2) + ... + (scalar^n-1 * point_n-1) 
+    fn test_scalar_point_algebraic_lc_combine<G: Group>(batching_scalar: G::ScalarField, points: Vec<G>) {
+        let points_ref = points.iter().collect::<Vec<_>>();
+
+        // Initialize LC and combine
+        let lc = LinearCombination::new_from_val(&batching_scalar, points_ref);
+        let combined_value = lc.combine();
+
+        // Compute the LC manually, instead, and check that the combined value is indeed the sum of the scalars times the points
+        let mut batching_scalar_pows = vec![G::ScalarField::one()];
+        (1..points.len())
+            .for_each(|i| {
+                batching_scalar_pows.push(batching_scalar_pows[i - 1] * &batching_scalar);
+            });
+        let expected_combined_value = batching_scalar_pows
+            .into_iter()
+            .zip(
+                points
+                    .into_iter()
+            )
+            .map(|(scalar, point)| point * &scalar)
+            .fold(G::zero(), |acc, val| acc + val);
+
+        assert_eq!(combined_value, expected_combined_value);
+    }
+
+    #[test]
+    fn test_scalar_point_lc() {
+        let rng = &mut thread_rng();
+
+        // Generate random scalars and points
+        let scalars = (0..MAX_LENGTH)
+            .map(|_| Fr::rand(rng))
+            .collect::<Vec<_>>();
+        let points = (0..MAX_LENGTH)
+            .map(|_| TweedleDee::rand(rng))
+            .collect::<Vec<_>>();
```

Comment:
Why not choosing a different parameter for the number of  samples? As I understand `MAX_LENGTH` is the length of a group vec, and typically a quite small number.

Reply:
It is set to 10 currently, it's good enough I think